### PR TITLE
Weather & evolution i18n with french translations

### PR DESCRIPTION
**Pull request by suriinam**
- Added Yes/No translation entries 
- Added evolution related translation entries
- Added weather related translation entries

### CODE REVIEW

File: src/locales/fr/menu.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { SimpleTranslationEntries } from "#app/plugins/i18n";
 
 export const menu: SimpleTranslationEntries = {
+  "yes": "Oui",
```
Comment (Flashfyre) [inline src/locales/fr/menu.ts:4]: Please add all new entries to all other languages as English.

**Review by Flashfyre — DISMISSED** [review]

File: src/locales/fr/menu.ts (hunk 1 above)
Comment (suriinam) [reply src/locales/fr/menu.ts:4]: @Flashfyre done !

**Review by Dakurei — CHANGES_REQUESTED** [review]
Missing spaces between certain punctuation marks in French
Especially all the words stuck to the "!" and "?"

**Comment by suriinam** [comment]
> Missing spaces between certain punctuation marks in French Especially all the words stuck to the "!" and "?"
True that in french a space before punctuation is mandatory but in the official Pokémon games it's not there...


What should we do ? :)

**Comment by Dakurei** [comment]
> https://github.com/pagefaultgames/pokerogue/assets/8146474/63e07fbd-015c-41bf-b3c3-826cdb904527

What should we do ? :)

**Review by Dakurei — APPROVED** [review]

**Comment by Dakurei** [comment]
I think some phrases have been translated literally, and don't match what's in the officials games (especially the harshSun that comes to mind), but this can be modified later if necessary 👍

**Comment by suriinam** [comment]
> No update here?

Updated

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**